db/seeds.d/050-taxonomies.rb

```diff
@@ -1,5 +1,5 @@
 # some associations shouldn't be set or require special handling.
-skip_associations = [:associated_audits, :audits, :default_users, :hosts,
+skip_associations = [:associated_audits, :audits, :default_users, :hosts, :discovered_hosts,
```

Comment:
I believe this should not be part of core and belongs to `discovery`. Sorry :-/

Reply:
How do we fix this though? I guess whitelisting wouldn't work either. We'd need to have some extendable list there and extend it from plugin. I wouldn't mind admitting in core. We have some plugins in situations like this, we have code specific to plugins in core templates too. We have discovery as one of the global pxe item in settings too. So I don't consider this as a terribly wrong. If you have some easy solution that I'm missing, I'm happy to hear that :-)

Reply:
Can this issue happen with other models defined in plugins? If so, we definitely need a generic solution.

Reply:
I defer to Tomer, but reading the whole seed file, I guess we have special handling for Hosts, because save on them can trigger trigger orchestration and typically, there's a lot of them.

Reply:
It could occur with other plugins if there are other objects that have associations to taxonomies, but i haven't found many other cases so far (other than templates, which are all skipped, and katello associations which wouldn't trigger this seed on upgrades since it already enforced enabling taxonomies). This PR came after a user reported that the seed was taking very long because they had thousands of discovered hosts during upgrade, each of them taking about 3 seconds.

Reply:
Ok. If that‘s the only use case I can live with this. ;-) Go ahead and thanks for your answers!